cmd/helmvm/token.go

```diff
@@ -0,0 +1,98 @@
+)
+
+var tokenCommands = &cli.Command{
+	Name:        "token",
+	Usage:       "Manage node join tokens",
+	Subcommands: []*cli.Command{tokenCreateCommand},
+}
+
+var tokenCreateCommand = &cli.Command{
+	Name:  "create",
+	Usage: "Creates a new node join token",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "role",
+			Usage: "The role of the token (can be controller or worker)",
+			Value: "worker",
+		},
+		&cli.DurationFlag{
+			Name:  "expiry",
+			Usage: "For how long the token should be valid",
+			Value: 24 * time.Hour,
+		},
+	},
+	Action: func(c *cli.Context) error {
+		if runtime.GOOS != "linux" {
+			return fmt.Errorf("token create is only supported on linux")
+		}
+		if os.Getuid() != 0 {
+			return fmt.Errorf("token create must be run as root")
+		}
+		role := c.String("role")
+		if role != "worker" && role != "controller" {
+			return fmt.Errorf("invalid role %q", role)
+		}
+		cfgpath := defaults.PathToConfig("k0sctl.yaml")
+		if _, err := os.Stat(cfgpath); err != nil {
+			if os.IsNotExist(err) {
+				logrus.Errorf("Unable to read the current cluster configuration.")
+				logrus.Errorf("Tried to read cluster config from %s", cfgpath)
+				logrus.Errorf("If you are running this command with sudo you may need")
+				logrus.Errorf("to run it as 'sudo -E' to preserve your user environment.")
+				return fmt.Errorf("k0s is not installed")
+			}
+			return fmt.Errorf("unable to stat k0sctl.yaml: %w", err)
+		}
+		logrus.Infof("Creating node join token for role %s", role)
+		if !defaults.DecentralizedInstall() {
+			logrus.Warn("You are opting out of the centralized cluster management.")
+			logrus.Warn("Through the centralized management you can manage all your")
+			logrus.Warn("cluster nodes from a single location. If you decide to move")
+			logrus.Warn("on the centralized management won't be available anymore")
+			question := &survey.Confirm{
+				Message: "Do you want to use continue ?", Default: true,
+			}
+			var moveOn bool
+			if err := survey.AskOne(question, &moveOn); err != nil {
+				return fmt.Errorf("unable to ask for user confirmation: %w", err)
+			} else if !moveOn {
+				return nil
+			}
+		}
+		dur := c.Duration("expiry").String()
+		buf := bytes.NewBuffer(nil)
+		cmd := exec.Command("k0s", "token", "create", "--expiry", dur, "--role", role)
+		cmd.Stdout = buf
+		cmd.Stderr = io.Discard
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("failed to create token: %w", err)
+		}
+		if !defaults.DecentralizedInstall() {
+			if err := defaults.SetInstallAsDecentralized(); err != nil {
+				return fmt.Errorf("failed to set decentralized install: %w", err)
+			}
+		}
+		logrus.Infof("Token created successfully.")
+		logrus.Infof("This token is valid for %s hours.", dur)
+		logrus.Infof("You can now run the following command in a remote node to add it")
+		logrus.Infof("to the cluster as a %q node:", role)
+		fmt.Printf("%s node join --role %q %s", defaults.BinaryName(), role, buf.String())
```

Comment:
would be cool to add a `--quiet, -q` flag to just print the token

Reply:
Can we use a log level instead?

Reply:
I think that what @emosbaugh is talking about here is to print only the token and not the full join command. Is that it @emosbaugh ?

Reply:
With all the logging happening here I am not sure this would be useful. At least not until we figure out what to do with the logging. :-(